### PR TITLE
Max client body size

### PR DESCRIPTION
**Pull request by a-levra**

### CODE REVIEW

**Comment by a-levra** [comment]
image

WOOOOOOOOW so beautiful

File: src/HttpMessages/HttpRequest.cpp [hunk 1]
```diff
@@ -42,7 +43,7 @@ HttpRequest::REQUEST_VALIDITY HttpRequest::checkValidity() {
 	std::map<enum LEXER_TOKENS, std::string> lexerToken;
 	ERRORS lexerValidity = autoLexer(lexerToken);
 	getLexerParserError(lexerValidity);
-	if (lexerValidity != ALL_LEXER_TOKENS_VALID){
+	if (lexerValidity != ALL_LEXER_TOKENS_VALID) {
 		_errors.push_back(lexerValidity);
 		coloredLog("Lexer error : ", getErrors(), RED);
```
Comment (tdameros) [inline src/HttpMessages/HttpRequest.cpp:48]: Replace it by logging

File: src/HttpMessages/HttpResponse.cpp [hunk 2]
```diff
@@ -31,52 +31,71 @@ HttpResponse &HttpResponse::operator=(const HttpResponse &other) {
...
+void HttpResponse::build() {
 
-
-	if ( !location.isAllowedMethod(_request.getMethod()) ){
-		buildErrorPage(405);
+	if ( !_location->isAllowedMethod(_request.getMethod()) ){
+		buildErrorPage(METHOD_NOT_ALLOWED);
 		return ;
 	}
 	coloredLog("Building...", "\"" + _request.getMethod() + "\"", BLUE);
```
Comment (tdameros) [inline src/HttpMessages/HttpResponse.cpp]: Replace it by logging

File: src/HttpMessages/HttpResponse.cpp [hunk 3]
```diff
@@ -91,26 +110,26 @@ void HttpResponse::buildGet(Location &location) {
...
 			this->buildErrorPage(404);
 			return ;
 		}
 	}
 	else
-		file = getFirstValidIndex(location);
+		file = getFirstValidIndex();
 
 	if (file == NULL){
 		coloredLog("No index valid: ", _requestUri, RED);
```
Comment (tdameros) [inline src/HttpMessages/HttpResponse.cpp]: Replace it by logging

File: src/HttpMessages/HttpResponse.cpp [hunk 4]
```diff
@@ -191,32 +207,47 @@ void HttpResponse::setHeader(const std::string& header, const std::string& conte
...
-		std::string pathToFile = location.getRoot() + location.getURI() + "/" + *it;
+		std::string pathToFile = _location->getRoot() + _location->getURI() + "/" + *it;
 		coloredLog("Index tested: ", pathToFile, YELLOW);
 		if (fileExists(pathToFile)){
 			coloredLog("Index found: ", pathToFile, GREEN);
 			return &(*it);
 		}
 	}
-	coloredLog("No index found: ", location.getURI(), RED);
+	coloredLog("No index found: ", _location->getURI(), RED);
```
Comment (tdameros) [inline src/HttpMessages/HttpResponse.cpp:223]: Same

File: src/HttpMessages/HttpResponse.cpp [hunk 5]
```diff
@@ -263,18 +294,22 @@ void HttpResponse::ExtractImgInsideBoundaries(std::string *boundary,
...
 	return resource;
 }
 
-const std::string *HttpResponse::tryGetFile(Location &location, const std::string &resource) {
-	std::string pathToFile = location.getRoot() + location.getURI() + "/" + resource;
+const std::string *HttpResponse::tryGetFile(const std::string &resource) {
+	if (!_location)
+		return NULL;
+	std::string pathToFile = _location->getRoot() + _location->getURI() + "/" + resource;
 	coloredLog("pathToFile: ", pathToFile, YELLOW);
```
Comment (tdameros) [inline src/HttpMessages/HttpResponse.cpp]: Replace it by logging

File: src/HttpMessages/HttpResponse.cpp [hunk 6]
```diff
@@ -263,18 +294,22 @@ void HttpResponse::ExtractImgInsideBoundaries(std::string *boundary,
...
 	if (fileExists(pathToFile))
 		return &resource;
 	coloredLog("File not found: ", pathToFile, RED);
```
Comment (tdameros) [inline src/HttpMessages/HttpResponse.cpp]: Same

File: src/server/Server.cpp [hunk 7]
```diff
@@ -81,7 +81,7 @@ VirtualServer *Server::getVirtualServer(const std::string &serverName) {
 }
 
 void Server::displayVirtualServers() {
-	coloredLog( "Webserv virtual servers(hosts): ", "", BLUE);
+	coloredLog("Webserv virtual servers(hosts): ", "", BLUE);
 	for (size_t i = 0; i < _virtualServers.size(); i++) {
 		coloredLog("\thost [" + toString(i) + "]: ", _virtualServers[i].getServerName()[0], PURPLE);
 	}
```
Comment (tdameros) [inline src/server/Server.cpp]: Replace coloredLog by logging

**Review by tdameros — CHANGES_REQUESTED** [review]

**Review by tdameros — APPROVED** [review]

**Pull request merged**